core/src/main/kotlin/fr/sncf/osrd/api/api_v2/pathfinding/PathfindingBlocksEndpointV2.kt

```diff
@@ -61,15 +61,12 @@ class PathfindingBlocksEndpointV2(private val infraManager: InfraManager) : Take
         } catch (error: NoPathFoundException) {
             pathfindingLogger.info("No path found")
             RsJson(RsWithBody(pathfindingResponseAdapter.toJson(error.response)))
-        } catch (error: OSRDError) {
-            if (!error.osrdErrorType.isCacheable) {
-                ExceptionHandler.handle(error)
-            } else {
-                pathfindingLogger.info("Pathfinding failed: ${error.message}")
-                val response = PathfindingFailed(error)
+        } catch (ex: Throwable) {
+            if (ex is OSRDError && ex.osrdErrorType.isCacheable) {
+                pathfindingLogger.info("Pathfinding failed: ${ex.message}")
+                val response = PathfindingFailed(ex)
                 RsJson(RsWithBody(pathfindingResponseAdapter.toJson(response)))
             }
-        } catch (ex: Throwable) {
             ExceptionHandler.handle(ex)
```

Comment:
If I understand correctly this modification, you only merged `catch (error: OSRDError)` with `catch (ex: Throwable)` only to factorize a single line which is `ExceptionHandler.handle(error)`. I'd say it was arguably better before because types were more specific in the branches, and the benefit of factorizing only one line might not worth it. But it's possible I missed something from this diff too. What do you think?

Reply:
You're right on what was done.
I was first mislead by the cognitive complexity while working in order to re-throw on `InfraLoadingSoftError` so I changed it 🤷
It's OK for me to change back if someone else votes for it 🙂

Reply:
There is behavior change here, in your change you call `ExceptionHandler.handle(ex)` unconditionally. I'm not sure this is what we want to do.

Reply:
Not sure how the `return try` handles "early returns"... So [I made it more explicit](https://github.com/OpenRailAssociation/osrd/pull/9439/commits/954a328b13a59d01c65b57848ec30f2190350bed).

Still, if you like better the previous implementation, I can revert (probably without the `return try`).

Reply:
OK, as discussed you had a good point, and it's corrected now.
Resolving this.